stl/src/filesystem.cpp

```diff
@@ -126,6 +126,39 @@ namespace {
         return __std_win_error{GetLastError()};
     }
 
+    [[nodiscard]] __std_win_error __stdcall _Get_file_id_by_handle(
+        const HANDLE _Handle, _Out_ FILE_ID_INFO* const _Id) noexcept {
+        __std_win_error _Last_error;
+        if (GetFileInformationByHandleEx(_Handle, FileIdInfo, _Id, sizeof(*_Id)) != 0) {
+            // if we could get FILE_ID_INFO, use that as the source of truth
+            return __std_win_error::_Success;
+        }
+
+        _Last_error = __std_win_error{GetLastError()};
+        switch (_Last_error) {
+        case __std_win_error::_Not_supported:
+        case __std_win_error::_Invalid_parameter:
+            break; // try more things
+        default:
+            return _Last_error; // real error, bail to the caller
+        }
+
+#ifndef _CRT_APP

```

Comment:
Can we simplify the else case to simply be `return __std_win_error{...}` and move the switch into here

Reply:
This is pre-existing code that's being extracted from `__std_fs_get_file_id()`. I see what you're suggesting, and the control flow transformation wouldn't be risky, but my general preference is to avoid significantly altering code when moving it (unless it's necessary). We can do this as a followup cleanup, since this doesn't affect correctness or performance.